### DISCUSSION TITLE
Tuya TS011F (polling) On/Off state not changed after manual button press

### DISCUSSION THREAD

**Discussion by FReaKaNDeLL** [Koenkk/zigbee2mqtt#9938] [Devices]
I have ordered some zigbee plugs and now that they arrived it seems to be the Tuya "TS011F_plug_3 TuYa Smart plug (with power monitoring by polling)". This plug is supported since 1.22.0 and it works as in, the sensors (power monitoring, volt, amp) are communicating every minute. I can control the on/off state (via HA or sending the MQTT manually). 

However, when i press the button on the plug, then it does not update the state. Weirdly when I connect a device and switch on the plug, the power monitoring is updated every minute, but the state is not updated at all.

the log shows this:
`Zigbee2MQTT:info  2021-12-01 21:41:11: MQTT publish: topic 'zigbee2mqtt/tuya_plug_1', payload '{"current":0,"energy":0,"linkquality":90,"power":0,"power_outage_memory":null,"state":"OFF","voltage":236}'`

I have tried to connect the plug with Conbee2 as well. This only gives me an on/off possibility and does not include the power monitoring values, however also via Conbee the on/off state is not updated when I press the button manually.

Is the state update simply not supported for this device?
Or can I try something to get it working?
I see in the `devices/tuya.js` file that indeed the state is not polled. Maybe here something can be done?

Any help is appreciated and I'm willing to tryout things.

**Comment by FReaKaNDeLL** [comment]
I have found on https://www.zigbee2mqtt.io/devices/TS011F_plug_3.html that i can poll the device manually. Is it possible to add this to the polling?

**Comment by Danit2** [comment]
See this: https://github.com/Koenkk/zigbee2mqtt/issues/9057

This is a known problem.
Theoretically a polling from on / off is possible but not useful.

**Reply by FReaKaNDeLL** [reply]
Thanks for your answer and letting me know it is a known problem.

For me it would be useful. Practically I need to be sure that the on/off state of all the devices I have connected are correct (what else is the point of a smart home?) So for me it would be useful to include the on/off state in the 60 second polling. I just assume that I do not look at the status in HA within less than 60 seconds after pressing the button manually (otherwise i would have switched it with HA).

So if this cannot be added (either as an option in the configuration.yaml or by adjusting the devices/tuya.js), then I feel the need to write a HA automation that runs every minute to request the state via MQTT.

In the meantime I found that I can also update the polling time using the configuration.yaml by adding measurement_poll_interval. It would be great the leave the user a choice to poll the state as well. A boolean like `also_poll_state: true` would be a great addition for this plug.

**Comment by Danit2** [comment]
You can use this external converter:
[TS011F_plug_3.zip](https://github.com/Koenkk/zigbee2mqtt/files/7640986/TS011F_plug_3.zip)

This will polling also the OnOff state.

**Comment by FReaKaNDeLL** [comment]
Thanks! I will try it tonight.

**Comment by FReaKaNDeLL** [comment]
It works perfectly.
Only downside that i see now is that it sends 3 MQTT messages in a row. This used to be only 2, and I think that I understand why. In the function which gets the data, there are 3 statements. It used to be 2.
```
                    try {
                        await endpoint.read('haElectricalMeasurement', ['rmsVoltage', 'rmsCurrent', 'activePower']);
                        await endpoint.read('seMetering', ['currentSummDelivered']);
                        await endpoint.read('genOnOff', ['onOff']);
                    } catch (error) {/* Do nothing*/}
```

Now I'm wondering if it is possible to gather all 3 `read` functions and send them out as 1 MQTT message. It's not urgent, it would be a nice to have. Any link to documentation is also fine, then i'll try to figure it out myself.

**Reply by RutgerDiehard** [reply]
I have 5 of these smart plugs with power monitoring (bought on Amazon and delivered from China). All of them work as expected in HA - power switching and power monitoring. However, they will all, at random times turn themselves off. This is not due to any automation in HA or by button press, they will just switch off. I did use an automation to turn them back on when switched off (which worked well) but they were bought to monitor tumble dryer/washing machines which don't enjoy being switched off and back on mid-cycle.

Have you seen this behaviour with yours? Does anyone know why they do this or how I stop them from doing it? They have never been used with a Tuya app/bridge.

**Reply by FReaKaNDeLL** [reply]
I have not seen this behaviour, however I only have 1 plug in use and not too often (trying to run a dehumidifier in the bathroom, which should not run too long anyway).

I understand your use case, as I have a Shelly plug on my laundry machine and dishwasher, which work perfectly, however they are wifi, which is less optimal.

What is your app version? (find it in the data/database.db and look for your plug's `appVersion`, which in my case is 69).

Reading your issue and knowing the quirks of my plugs, this is most likely the last time I bought a Tuya device. I also had issues with a Tuya wifi plug and this is not very promising. However it can also be an issue with the relays inside your plugs. I have one Shelly 1 which has a faulty relay, which pops back once in a while. But this only happens directly after switching on and not randomly when it is already on for a while.
